src/containerapp/azext_containerapp/custom.py

```diff
@@ -2413,3 +2468,276 @@ def delete_environment_telemetry_app_insights(cmd,
+
+    try:
+        r = containerapp_env_telemetry_otlp_decorator.show()
+    except CLIError as e:
+        handle_raw_exception(e)
+
+    existing_otlps = safe_get(r, "properties", "openTelemetryConfiguration", "destinationsConfiguration", "otlpConfigurations")
+    if existing_otlps is not None:
+        otlp = [p for p in existing_otlps if p["name"].lower() == otlp_name.lower()]
+
+        if not otlp:
+            raise ValidationError(f"Otlp entry with name {otlp_name} does not exist, please make sure the {otlp_name} already added")
+
+    containerapp_env_telemetry_otlp_decorator.construct_payload()
+    r = containerapp_env_telemetry_otlp_decorator.update()
+    
+    return r
+
+
+def remove_environment_telemetry_otlp(cmd,
+                                      name,
+                                      resource_group_name,
+                                      otlp_name,
+                                      no_wait=False):
+    raw_parameters = locals()
+    raw_parameters["enable_open_telemetry_traces"] = False
+    raw_parameters["enable_open_telemetry_logs"] = False
+    raw_parameters["enable_open_telemetry_metrics"] = False
+    containerapp_env_telemetry_otlp_decorator = ContainerappEnvTelemetryOtlpPreviewSetDecorator(
+        cmd=cmd,
+        client=ManagedEnvironmentPreviewClient,
+        raw_parameters=raw_parameters,
+        models=CONTAINER_APPS_SDK_MODELS
+    )
+    containerapp_env_telemetry_otlp_decorator.register_provider(CONTAINER_APPS_RP)
+
+    containerapp_env_telemetry_otlp_decorator.construct_remove_payload()
+    r = containerapp_env_telemetry_otlp_decorator.update()
+    
+    return r
+
+def show_environment_telemetry_otlp(cmd,
+                                    name,
+                                    resource_group_name,
+                                    otlp_name):
+    raw_parameters = locals()
+
+    containerapp_env_telemetry_otlp_decorator = ContainerappEnvTelemetryOtlpPreviewSetDecorator(
+        cmd=cmd,
+        client=ManagedEnvironmentPreviewClient,
+        raw_parameters=raw_parameters,
+        models=CONTAINER_APPS_SDK_MODELS
+    )
+    
+    containerapp_env_def = None
+    try:
+        containerapp_env_def = containerapp_env_telemetry_otlp_decorator.show()
+    except CLIError as e:
+        handle_raw_exception(e)
+
+    if not containerapp_env_def:
+        raise ResourceNotFoundError("The containerapp environment '{}' does not exist".format(name))
+
+    existing_otlps = safe_get(containerapp_env_def, "properties", "openTelemetryConfiguration", "destinationsConfiguration", "otlpConfigurations")
+    if existing_otlps is not None:
+        otlp = [p for p in existing_otlps if p["name"].lower() == otlp_name.lower()]
+
+        if not otlp:
+            raise ValidationError(f"Otlp entry with name {otlp_name} does not exist, please retry with different name")
+                
+        r = {}
+
+        if "headers" in otlp[0]:
+            dict = otlp[0]["headers"]
```

Comment:
About the var name `dict`, there is a warning: Shadows built-in name 'dict' .
Please consider using another name.